ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -855,13 +876,21 @@ withTestMempool setup@TestSetup {..} prop =
                            , snapshotSlotNo
                            } =
         case runExcept $ repeatedlyM
-               (fmap (TickedSimpleLedgerState . convertMapKind . getTickedSimpleLedgerState  . fst) .: applyTx testLedgerConfig DoNotIntervene snapshotSlotNo)
-               txs
-               (TickedSimpleLedgerState (convertMapKind ledgerState)) of
+               (\tx st -> do
+                   st' <- applyTx testLedgerConfig DoNotIntervene snapshotSlotNo tx (forgetLedgerTablesDiffsTicked st)
+                   pure (zipOverLedgerTablesTicked g (fst st') (projectLedgerTablesTicked st)))
+               [ txForgetValidated tx | (tx, _) <- snapshotTxs ]
+               (mapOverLedgerTablesTicked f $ TickedSimpleLedgerState ledgerState) of
```

Comment:
Considering the size of the case expression, maybe we should make a local binding for it

Reply:
Yeah, maybe it's worth abstracting away the first arg of `repeatedlyM` and adding some types to said abstraction.